src/tiledb/cloud/bioimg/exportation.py

```diff
@@ -64,36 +76,66 @@ def build_io_uris_exportation(
 
         vfs = tiledb.VFS()
 
-        def create_output_path(input_file, output_dir) -> str:
-            filename = os.path.splitext(os.path.basename(input_file))[0]
-            output_filename = filename + f".{output_ext}" if output_ext else filename
-            return os.path.join(output_dir, output_filename)
-
-        def iter_paths(source: Sequence) -> Iterator[Tuple]:
-            for uri in source:
-                if vfs.is_dir(uri) and tiledb.object_type(uri) != "group":
-                    # Folder for exploration
-                    contents = vfs.ls(uri)
-                    yield from iter_paths(contents)
-                elif tiledb.object_type(uri) == "group":
-                    # For exportation we require the source path to be a tiledb group
-                    yield uri, create_output_path(uri, output_dir)
-
-        if len(source) == 0:
-            raise ValueError("The source files cannot be empty")
-        return tuple(iter_paths(source))
+        def create_output_path(input_file: str, output: str) -> str:
+            # Check if output is dir
+            if output.endswith("/"):
+                filename = os.path.splitext(os.path.basename(input_file))[0]
+                output_filename = (
+                    filename + f".{output_ext}" if output_ext else filename
+                )
+                return os.path.join(output, output_filename)
+            else:
```

Comment:
no else after an unconditional return. you might rework this function as

```python
if not output.endswith("/"):
  return output
# the rest
```

just so most of the code isn’t indented